src/CodeGen_LLVM.cpp

```diff
@@ -2352,12 +2437,30 @@ llvm::Value *CodeGen_LLVM::codegen_dense_vector_load(const Type &type, const std
         Value *elt_ptr = codegen_buffer_pointer(name, type.element_of(), slice_base);
         Value *vec_ptr = builder->CreatePointerCast(elt_ptr, slice_type->getPointerTo());
 
-        Instruction *load_inst;
-        if (vpred != nullptr) {
-            Value *slice_mask = slice_vector(vpred, i, slice_lanes);
-            load_inst = builder->CreateMaskedLoad(slice_type, vec_ptr, llvm::Align(align_bytes), slice_mask);
+        Value *slice_mask = (vpred != nullptr) ? slice_vector(vpred, i, slice_lanes) : nullptr;
+
+        Instruction *load_inst = nullptr;
+        if (stride) {
```

Comment:
I'm confused. Based on the calling code, stride is non-null if it's a compile-time constant, which means we're calling the strided load intrinsic when the stride is the constant 1 (i.e. a dense vector load), and we're calling llvm.vp.load otherwise (e.g. a load vectorized across y)?

Reply:
I modified the calling location to not do this for the dense case. Ideally it doesn't matter as LLVM should strength reduce the vector predication instructions for constant arguments, but some experiments should it isn't very good about it for constant one masks and constant length that matches the fixed SIMD width on e.g. x86.